traffic_portal/app/src/common/modules/table/profilesParamsCompare/TableProfilesParamsCompareController.js

```diff
@@ -0,0 +1,213 @@
+						)
+						.finally(
+							function() {
+								let promises = [];
+
+								if (updateProfile1) {
+									promises.push(profileParameterService.linkProfileParameters(profile1, _.pluck($scope.selectedProfile1Params, 'id')));
+								}
+
+								if (updateProfile2) {
+									promises.push(profileParameterService.linkProfileParameters(profile2, _.pluck($scope.selectedProfile2Params, 'id')));
+								}
+
+								if (promises.length > 0) {
+									$q.all(promises)
+										.then(
+											function(result) {
+												let messages = [];
+												for (let i = 0; i < result.length; i++) {
+													messages = _.union(messages, result[i].data.alerts);
+												};
+												messageModel.setMessages(messages, false);
+											})
+										.finally(
+											function() {
+												$scope.refresh();
+											}
+										);
+								}
+							}
+						);
+				}
+			);
+	};
+
+	$scope.dirty = false;
+
+	$scope.profile1 = profile1;
+	$scope.profile2 = profile2;
+	$scope.profilesParams = profilesParams;
+
+	$scope.selectedProfile1Params = [];
+	$scope.selectedProfile2Params = [];
+
+	$scope.selectedParams = _.map(profilesParams, function(param) {
+		let isAssignedToProfile1 = _.find(profile1.params, function(profile1param) { return profile1param.id == param.id }),
+			isAssignedToProfile2 = _.find(profile2.params, function(profile2param) { return profile2param.id == param.id });
+
+		if (isAssignedToProfile1) {
+			param['origSelected1'] = true;
+			param['selected1'] = true;
+		} else {
+			param['origSelected1'] = false;
+			param['selected1'] = false;
+		}
+
+		if (isAssignedToProfile2) {
+			param['origSelected2'] = true;
+			param['selected2'] = true;
+		} else {
+			param['origSelected2'] = false;
+			param['selected2'] = false;
+		}
+
+		return param;
+	});
+
+	$scope.updateSelectedCount = function(profNum) {
+		$scope['selectedProfile' + profNum + 'Params'] = _.filter($scope.selectedParams, function(param) { return param['selected' + profNum] == true; } );
+	};
+
+	$scope.isCheckboxDirty = function(pp, profNum) {
+		return pp['selected' + profNum] !== pp['origSelected' + profNum];
+	};
+
+	$scope.onChange = function(profNum) {
+		$scope.dirty = true;
+		$scope.updateSelectedCount(profNum);
+	};
+
+	$scope.refresh = function() {
+		$state.reload(); // reloads all the resolves for the view
+	};
+
+	$scope.restore = function() {
+		let params = {
+			title: 'Restore Parameter Assignments?',
+			message: 'Any changes you have made will be lost and the original parameter assignments will be restored.'
+		};
+		let modalInstance = $uibModal.open({
+			templateUrl: 'common/modules/dialog/confirm/dialog.confirm.tpl.html',
+			controller: 'DialogConfirmController',
+			size: 'md',
+			resolve: {
+				params: function () {
+					return params;
+				}
+			}
+		});
+		modalInstance.result.then(function() {
+			$scope.refresh();
+		});
+	};
+
+	$scope.update = function() {
+		confirmUpdateProfiles();
```

Comment:
I think you can actually just do
```javascript
$scope.update = confirmUpdateProfiles;
```

Reply:
good point. i think i had more stuff in there at one time.